stream.go

```diff
@@ -14,6 +14,24 @@ import (
 	"github.com/lucas-clemente/quic-go/internal/wire"
 )
 
+type streamI interface {
```

Comment:
I'm strongly opposed to making the internal stream methods public (even if invisible). This way they can easily (without pointer magic) be called from external code – and might even break if the stdlib introduces more interfaces in the future.

Also, that change should really be its own commit if at all possible.

Reply:
Making those methods public is the only way to mock them. As it was before, we were just inconsistent (for example, `AddStreamFrame` was public already).
I'm not sure I understand your comment about the stdlib.

Separating this out into a separate commit is challenging, because this will involve changing the tests as well (which used to be tightly coupled to the flow control stuff).

Reply:
What we can do is create a separate `Stream` struct on which we **only** define the methods defined on the interface, e.g.

```go
type Stream struct {
  stream streamI
}

func (s *Stream) Read(b []byte) (int, error) {
  return s.stream.Read(b)
}

// ... all other Stream interface methods
```

That would allow us to use a mock for `streamI` while at the same time preventing users to type-assert the public methods on the `quic.Stream`.

Reply:
This is not a mockgen problem here. mockgen generates the right mock calls here, but Go doesn't allow private methods on an implementation that lives in another package.